hazelcast/src/main/java/com/hazelcast/config/tpc/TPCSocketConfig.java

```diff
@@ -0,0 +1,49 @@
+package com.hazelcast.config.tpc;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+public class TPCSocketConfig {
+    private static final Collection<String> DEFAULT_PORT_DEFINITIONS = Collections.singleton("11000-21000");
+    private static final int DEFAULT_RECEIVE_BUFFER_SIZE = 128 * 1024;
+    private static final int DEFAULT_SEND_BUFFER_SIZE = 128 * 1024;
+
+    private Collection<String> portDefinitions = DEFAULT_PORT_DEFINITIONS;
+    private int receiveBufferSize = DEFAULT_RECEIVE_BUFFER_SIZE;
+    private int sendBufferSize = DEFAULT_SEND_BUFFER_SIZE;
+
+    public Collection<String> getPortDefinitions() {
+        return portDefinitions;
+    }
+
+    public TPCSocketConfig addPortDefinition(@Nonnull String portDefinition) {
```

Comment:
Should there be any syntax validation? Or duplicate? What if there are multiple overlapping entries?

Reply:
Overlapping entries isn't a problem. See how `AdddressUtil.transformPortDefinitionsToPorts()` works.

https://github.com/hazelcast/hazelcast/pull/23254/commits/9f6c9bbbe7c89f3d3257eb7f23779d915f910a4f

Reply:
BTW, I used a simple pattern to validate here. However `AdddressUtil.transformPortDefinitionsToPorts()` is actually more powerful. For example it can parse the following list:
```
{"12000,12500-12600 13000", "14000-1500; 16000-17000 18000"}
```
However, giving users ability to configure like this is confusing IMO. That is why I used a simple pattern. If you think otherwise let me know.

To elaborate more, the other code part this method used is outbound-ports. They can be configured like this:
```
outbound-ports:
  - 33000-35000
  - 37000,37001,37002,37003
  - 38000,38500-38600
```
However tpc port definitions can only be configured like this currently (because of added validation):
```
port-definitions:
  - 11000-12000
  - 16000-18000
```

Reply:
https://github.com/hazelcast/hazelcast/pull/23254/commits/e86bef401d3839f0a5414385a01cae913f05a919